scripts.js

```diff
@@ -91,35 +110,8 @@ window.blog = function() {
     FR: 'fr',
   };
   const context = '/';
-  let language = LANG.EN;
-  let pageType = TYPE.HOME;
-  const segs = window.location.pathname
-    .split('/')
-    .filter(seg => seg !== '');
-  if (segs.length > 0) {
-    if (segs.length >= 1) {
-      // language
-      for (let [key, value] of Object.entries(LANG)) {
-        if (value === segs[0]) {
-          language = value;
-          break;
-        }
-      }
-    }
-    if (segs.length >= 2) {
-      // post pages
-      if (segs[1] === 'drafts' || segs[1] === 'publish' || segs[1] === 'documentation' || /\d{4}\/\d{2}\/\d{2}/.test(segs.join('/'))) {
-        pageType = TYPE.POST;
-      } else {
-        for (let [key, value] of Object.entries(TYPE)) {
-          if (segs[1].startsWith(value)) {
-            pageType = value;
-            break;
-          }
-        }
-      }
-    }
-  }
+  const language = LANG[getDetectedLanguage().toUpperCase()] || LANG.EN;
+  const pageType = TYPE[getDetectedPageType().toUpperCase()] || TYPE.HOME;
```

Comment:
I would move the `LANG`/`TYPE` lookup and defaults into the getters for more encapsulation and robustness, so these lines would read:
```suggestion
  const language = getDetectedLanguage();
  const pageType = getDetectedPageType();
```